manage.py

```diff
@@ -145,6 +146,272 @@ def refresh_materialized(concurrent=True):
     logger.info("Finished refreshing materialized views.")
 
 
+def create_public_api_key(
+        space,
+        first_rate_limit,
+        first_rate_limit_duration,
+        second_rate_limit,
+        second_rate_limit_duration):
+
+    logger.info("Creating new public API key for {} environment".format(space))
+    UMBRELLA_ADMIN_AUTH_TOKEN = env.get_credential("UMBRELLA_ADMIN_AUTH_TOKEN")
+    API_KEY = env.get_credential("FEC_WEB_API_KEY_PUBLIC")
+
+    header = {
+        "X-Admin-Auth-Token": UMBRELLA_ADMIN_AUTH_TOKEN,
+        "X-Api-Key": API_KEY,
+        "Content-Type": "application/json; charset=UTF-8",
+    }
+
+    create_api_key_params = {
+        "first_name": space,
+        "last_name": "Public API Key",
+        "email": env.get_credential("FEC_EMAIL"),
+        "use_description": "FEC_WEB_API_KEY_PUBLIC for {} environment. Rate limited key per IP. Created {}".format(
+            space,
+            datetime.datetime.today()),
+        "registration_source": "update_public_api_key task",
+        "throttle_by_ip": True,
+        "terms_and_conditions": True,
+        "created_at": datetime.datetime.now().isoformat(),
+        "creator": {
+            "username": "auto-generated"
+        },
+        "settings": {
+            "rate_limit_mode": "custom",
+            "rate_limits": [{
+                "limit_by": "apiKey",
+                "response_headers": True,
+                "limit": first_rate_limit,
+                "duration": first_rate_limit_duration
+                },
+                {
+                "limit_by": "apiKey",
+                "response_headers": False,
+                "limit": second_rate_limit,
+                "duration": second_rate_limit_duration,
+                }
+            ]
+        }
+    }
+
+    url = "https://api.data.gov/api-umbrella/v1/users.json"
+
+    response = requests.post(url, json=create_api_key_params, headers=header)
+
+    try:
+        response.raise_for_status()
+    except requests.exceptions.HTTPError as error:
+        error_message = "Error occured when creating API key, check logs"
+        slack_message(error_message)
+        logger.error("Error occured with creating API key: {}".format(error))
+        raise
+
+    response_json = response.json()
+    new_api_key = response_json["user"]["api_key"]
+
+    logger.info("New API key: {}".format(new_api_key))
```

Comment:
@tmpayton Could you implement a split function or similar method to retrieve only the initial 5 characters of the newly generated API key for logging purposes, for security concerns, rather than displaying the entire 40-character key?